### PR TITLE
Feature/validation once

### PR DESCRIPTION
**Pull request by aiorhiroki**
# 高速化：Callbackで推論しないようにした

- Tensorflowのmetricで、クラス別のDice,IoUを出せるようにした。
- PlotManagerを修正し、出せるログは全て自動で取得してPlotできるようにした。

### CODE REVIEW

File: farmer/ncc/callbacks/keras_callbacks.py [hunk 1]
```diff
@@ -144,25 +144,26 @@ def on_epoch_end(self, epoch, logs={}):
...
-                labels=[
-                    "train",
-                    "validation"
-                ]
-            )
+    def __init__(self, save_dir):
+        self.plot_manager = MatPlotManager(save_dir)
 
     def on_epoch_end(self, epoch, logs={}):
+        if epoch == 0:
```
Comment (yumion) [inline farmer/ncc/callbacks/keras_callbacks.py:262]: ここon_train_beginじゃなくてif文にしたのはなぜですか？

File: farmer/ncc/callbacks/keras_callbacks.py [hunk 2]
```diff
@@ -144,25 +144,26 @@ def on_epoch_end(self, epoch, logs={}):
...
+                    xy_labels=("epoch", metric),
+                    labels=[
+                        "train",
+                        "validation"
+                    ]
+                )
         # update learning figure
-        for metric in self.metrics:
+        for metric in logs.keys():
+            if metric.startswith("val_"):
```
Comment (yHarai) [inline farmer/ncc/callbacks/keras_callbacks.py:276]: ここではvalをスキップして、
下記のコードではvalを参照しているのは何か違いがあるのでしょうか？
https://github.com/aiorhiroki/farmer.tf2/pull/8/files#diff-8a91595173f5bf1d9b9c375c09f4e40c089ae2d2142fb76fe8cbce152ccc2b12L237

File: farmer/ncc/callbacks/keras_callbacks.py (hunk 1 above)
Comment (aiorhiroki) [reply farmer/ncc/callbacks/keras_callbacks.py:262]: on_train_beginはlogs.keysが空だからです！
最初のエポックでlogs.keysの中身を見て、その後plotするものを自動で取得する実装になってる。

File: farmer/ncc/callbacks/keras_callbacks.py (hunk 2 above)
Comment (aiorhiroki) [reply farmer/ncc/callbacks/keras_callbacks.py:276]: logs.keys()には["learning_rate", "loss", "val_loss", "iou", "val_iou"]などが詰め込まれていて、
例えば、"lossとval_loss"、"iouとval_iou"をセットに同じファイルにプロットしたいための実装になってます。
最初のスキップでvalがついてないやつ（例えばloss）を取り出して、次のfigure_addでlossとval_lossをセットでプロットします。

**Comment by yumion** [comment]
conflict解消をお願いします！
そしたらLGTMです！！:+1::+1:
